programs/native/storage/src/lib.rs

```diff
@@ -46,8 +187,41 @@ fn entrypoint(
 #[cfg(test)]
 mod test {
     use super::*;
-    use solana_sdk::account::create_keyed_accounts;
-    use solana_sdk::signature::{Keypair, KeypairUtil};
+    use solana_sdk::account::{create_keyed_accounts, Account};
+    use solana_sdk::hash::Hash;
+    use solana_sdk::signature::{Keypair, KeypairUtil, Signature};
+    use solana_sdk::storage_program::ProofStatus;
+    use solana_sdk::storage_program::StorageTransaction;
+    use solana_sdk::transaction::{Instruction, Transaction};
+
+    fn test_transaction(
+        tx: &Transaction,
+        program_accounts: &mut [Account],
+    ) -> Result<(), ProgramError> {
+        assert_eq!(tx.instructions.len(), 1);
+        let Instruction {
+            ref accounts,
+            ref userdata,
+            ..
+        } = tx.instructions[0];
```

Comment:
If you borrow `tx.instructions[0]`, you don't need to do that awkward pattern match:

```rust
let ix = &tx.instructions[0];
info!("accounts: {:?}", ix.accounts);
```